src/components/things/names.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Class used to map keys to string literals
+ */
+export class Names {
+    public readonly AreaGate: string = "AreaGate";
+    public readonly BubbleLarge: string = "BubbleLarge";
+    public readonly BubbleSmall: string = "BubbleSmall";
+    public readonly Exclamation: string = "Exclamation";
+    public readonly ExplosionLarge: string = "ExplosionLarge";
+    public readonly ExplosionSmall: string = "ExplosionSmall";
+    public readonly FenceWide: string = "FenceWide";
+    public readonly HealingMachineBall: string = "HealingMachineBall";
+    public readonly NIDORINOFront: string = "NIDORINOFront";
+    public readonly Note: string = "Note";
+    public readonly Oak: string = "Oak";
+    public readonly OakPortrait: string = "OakPortrait";
+    public readonly Player: string = "Player";
+    public readonly PlayerPortrait: string = "PlayerPortrait";
+    public readonly PlayerSilhouetteLarge: string = "PlayerSilhouetteLarge";
+    public readonly PlayerSilhouetteSmall: string = "PlayerSilhouetteSmall";
+    public readonly Rival: string = "Rival";
+    public readonly RivalPortrait: string = "RivalPortrait";
+    public readonly ScratchLine: string = "ScratchLine";
+    public readonly shadow: string = "shadow";
+    public readonly SightDetector: string = "SightDetector";
+    public readonly Square: string = "Square";
+    public readonly WhiteSquare: string = "WhiteSquare";
```

Comment:
Other files that have constants, even ones that are normally PascalCase (like `"OakPortrait"`) typically use camelCase for the names (which would make that `.oakPortrait` pointing to `"OakPortrait"`). Let's keep that consistent in this file.